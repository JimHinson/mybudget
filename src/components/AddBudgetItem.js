import React, { useState } from "react";
import "../styles/monthly.css";
import "../styles/App.css";
import { Button, Segment, Form, Input } from "semantic-ui-react";
import "semantic-ui-css/semantic.min.css";

function AddBudgetItem(props) {
  const [budgetItem, setbudgetItem] = useState("");
  const [amount, setamount] = useState(0);

  const handleSubmit = (event)=> { 
    event.preventDefault(); 
    setamount(0)
    setbudgetItem("")
    props.handleSubmit(budgetItem, amount)
  } 
 


  return (
    <form onSubmit={handleSubmit} className="formBody">
      <Input
        type="text"
        id="budgetItem"
        className="input"
        value={budgetItem}
        onChange={e => setbudgetItem(e.target.value)}
        placeholder="budget item"
        required
      />

      <Input
        type="text"
        id="amount"
        className="input"
        value={amount}
        onChange={e => setamount(e.target.value)}
        placeholder="amount"
        icon='dollar sign' iconPosition='left'
        pattern="[0-9.]+"
        required
        
      />
      {/* <button className="myBtn" type="submit">Submit</button> */}
      <Button primary type="submit" data-selected="true" data-label-id="0">Submit</Button>
    </form>
  );
}


export default AddBudgetItem;
