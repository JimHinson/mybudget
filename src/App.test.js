
import React from 'react';
import { shallow } from 'enzyme';
import AddBudgetItem from '../src/components/AddBudgetItem';

describe('<AddBudgetItem /> with no props', () => {
  const container = shallow(<AddBudgetItem />);
  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  });

  // it('should have an email field', () => {
  //   expect(container.find('input[type="email"]').length).toEqual(1);
  // });

  // it('should have proper props for email field', () => {
  //   expect(container.find('input[type="email"]').props()).toEqual({
  //     className: 'mx-auto my-2',
  //     onBlur: expect.any(Function),
  //     placeholder: 'email',
  //     type: 'email',
  //   });
  });