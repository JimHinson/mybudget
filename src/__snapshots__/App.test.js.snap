// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddBudgetItem /> with no props should match the snapshot 1`] = `"<form class=\\"formBody\\" data-tut=\\"reactour__transactionList\\"><div style=\\"border:solid 2px black;border-radius:9px\\" class=\\"ui input input\\"><input type=\\"text\\" id=\\"budgetItem\\" value=\\"\\" placeholder=\\"budget item\\" required=\\"\\"/></div><div style=\\"border:solid 2px black;border-radius:9px\\" class=\\"ui left icon input input\\"><input type=\\"text\\" id=\\"amount\\" value=\\"0\\" placeholder=\\"amount\\" pattern=\\"[0-9.]+\\" required=\\"\\"/><i aria-hidden=\\"true\\" class=\\"dollar sign icon\\"></i></div><button type=\\"submit\\" data-selected=\\"true\\" data-label-id=\\"0\\" class=\\"ui primary button\\">Submit</button></form>"`;
